code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -195,6 +195,87 @@
 	direct_hit_shrapnel = 5
 	dispersion_angle = 360 //beeg circle
 
+/*
+//================================================
+				M203 Grenades
+//================================================
+*/
+
+/obj/item/explosive/grenade/incendiary/impact
```

Comment:
consider converting the `impact` functionality into an element/component for grenades, instead of copy-pasting the same code across grenade types